features/device-config/persistence/api/src/main/java/org/opennms/features/deviceconfig/persistence/api/DeviceConfig.java

```diff
@@ -181,12 +186,22 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (!(o instanceof DeviceConfig)) return false;
         DeviceConfig that = (DeviceConfig) o;
-        return ipInterface.equals(that.ipInterface) && Arrays.equals(config, that.config) && encoding.equals(that.encoding) && configType == that.configType && Objects.equals(failureReason, that.failureReason) && Objects.equals(createdTime, that.createdTime) && lastUpdated.equals(that.lastUpdated) && Objects.equals(lastFailed, that.lastFailed) && Objects.equals(lastSucceeded, that.lastSucceeded);
+        return
```

Comment:
If you use IntelliJ default generation, changes wouldn't look much different

Reply:
Done

Reply:
Updated to remove `id` field, fixed ordering so that `ipInterface` and `config` are done last, so usually they won't need to be checked (expensive check).